bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,320 @@
+
+#[wasm_bindgen(js_class = Account)]
+impl WasmAccount {
+  #[wasm_bindgen(js_name = did)]
+  pub fn did(&self) -> WasmDID {
+    let account: Ref<Account> = self.0.as_ref().borrow();
+    WasmDID::from(account.document().id().clone())
+  }
+
+  /// Returns whether auto-publish is enabled.
+  #[wasm_bindgen]
+  pub fn autopublish(&self) -> bool {
+    self.0.as_ref().borrow().autopublish()
+  }
+
+  /// Returns the auto-save configuration value.
+  #[wasm_bindgen]
+  pub fn autosave(&self) -> WasmAutoSave {
+    WasmAutoSave(self.0.as_ref().borrow().autosave())
+  }
+
+  /// Returns a copy of the document managed by the `Account`.
+  #[wasm_bindgen]
+  pub fn document(&self) -> WasmDocument {
+    let document: IotaDocument = self.0.as_ref().borrow().document().clone();
+    WasmDocument::from(document)
+  }
+
+  /// Resolves the DID Document associated with this `Account` from the Tangle.
+  #[wasm_bindgen(js_name = resolveIdentity)]
+  pub fn resolve_identity(&self) -> PromiseResolvedDocument {
+    let account = self.0.clone();
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow()
+        .resolve_identity()
+        .await
+        .map(WasmResolvedDocument::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+    promise.unchecked_into::<PromiseResolvedDocument>()
+  }
+
+  /// Removes the identity from the local storage entirely.
+  ///
+  /// Note: This will remove all associated document updates and key material - recovery is NOT POSSIBLE!
+  #[wasm_bindgen(js_name = deleteIdentity)]
+  pub fn delete_identity(self) -> Promise {
+    // Get IotaDID and storage from the account.
+    let account: Rc<WasmRefCell<Account>> = self.0;
+    let did: IotaDID = account.as_ref().borrow().did().to_owned();
+    let storage: Arc<dyn Storage> = Arc::clone(account.as_ref().borrow().storage());
+
+    // Drop account should release the DIDLease because we cannot take ownership of the Rc.
+    // Note that this will still fail if anyone else has a reference to the Account.
+    std::mem::drop(account);
+
+    future_to_promise(async move {
+      // Create a new account since `delete_identity` consumes it.
+      let account: Result<Account> = AccountBuilder::new()
+        .storage(AccountStorage::Custom(storage))
+        .load_identity(did)
+        .await
+        .wasm_result();
+
+      match account {
+        Ok(a) => a.delete_identity().await.wasm_result().map(|_| JsValue::undefined()),
+        Err(e) => Err(e),
+      }
+    })
+  }
+
+  /// Push all unpublished changes to the tangle in a single message.
+  #[wasm_bindgen]
+  pub fn publish(&mut self, publish_options: Option<WasmPublishOptions>) -> Promise {
+    let account = self.0.clone();
+    let mut options: PublishOptions = PublishOptions::default();
+    if let Some(publish_options) = publish_options {
+      options = PublishOptions::from(publish_options);
+    };
+    future_to_promise(async move {
```

Comment:
Nit: `PublishOptions` implements `Default` so this can be condensed somewhat. Should not be mutable either way even with `if-let` syntax.

```suggestion
    let options: PublishOptions = publish_options.map(PublishOptions::from).unwrap_or_default();
    
    let account: Rc<WasmRefCell<Account>> = Rc::clone(&self.0);
    future_to_promise(async move {
```